src/components/fields/checkbox-group/types.ts

```diff
@@ -10,6 +10,7 @@ interface IOption {
 
 interface IToggleOption extends IOption {
 	none?: boolean;
+	children?: Record<string, TFrontendEngineFieldSchema>;
```

Comment:
- add `undefined` typing
- I just noticed the toggle typing is not applied correctly, can we differentiate the option typing for toggle styleType? so it will reject children if styleType is default

```suggestion
	none?: boolean | undefined;
	children?: Record<string, TFrontendEngineFieldSchema> | undefined;
```

Reply:
> I just noticed the toggle typing is not applied correctly, can we differentiate the option typing for toggle styleType? so it will reject children if styleType is default

after some experimentation, found that the `styleType` is nested, so Typescript is not able to perform type narrowing with discriminated union

I've also tried:
- set options to `IToggleOption[] | IOption[]` - not that helpful
- define the additional options like `customOptions.additionalOptions: IToggleOption[]` but that would decouple it from the main options

Reply:
I experimented a bit and found a way to make it work
- the schema for each styleType will need to be defined separately
- the final schema will be a type and not an interface anymore
```
interface ICheckboxGroupToggleSchema<V = undefined>
	extends IBaseFieldSchema<"checkbox", V>,
		TComponentOmitProps<CheckboxProps> {
	options: IToggleOption[];
	customOptions?:
		| {
				styleType: "toggle";
				indicator?: boolean | undefined;
				border?: boolean | undefined;
				layoutType?: TCheckboxToggleLayoutType | undefined;
		  }
		| undefined;
}

interface ICheckboxGroupDefaultSchema<V = undefined>
	extends IBaseFieldSchema<"checkbox", V>,
		TComponentOmitProps<CheckboxProps> {
	options: IOption[];
	customOptions?:
		| {
				styleType: "default";
		  }
		| undefined;
}

export type TCheckboxGroupSchema<V = undefined> = ICheckboxGroupToggleSchema<V> | ICheckboxGroupDefaultSchema<V>;
```

Reply:
oh yes, this works. thanks for looking into this too

will the renaming of the interface be a breaking change?

Reply:
yeah I found that some projects are referencing the schema typing so it is going to break some projects

in this case, let's retain the old names yet change them to types. we will mark them as deprecated, I will announce their deprecation in the upcoming release and remove it in a future release

for the new typings, we will use them in place of the existing ones in `TFieldSchema`